source/lib/JMSLab/autofill.py

```diff
@@ -32,6 +32,15 @@ def GenerateAutofillMacros(autofill_lists,  autofill_formats, autofill_outfile):
     .tex file
 
     """
+    if type(autofill_lists) != list:
+            raise Exception("Argument 'autofill_lists' must be list")
+            
+    nested_list = any(isinstance(i, list) for i in autofill_lists)
```

Comment:
I'm not sure in what context we may want `autofill_lists` to be a nested list?

Reply:
@santiagohermo sorry that it's not very clear from the get go, but the original use case that this function emerged from was something along the lines of this:
`
GenerateAutofillMacros([["Epsilon", "MarginalCost"],["CartelProfits", "CartelCosts"]], 
["{:.2f}", "\\textnormal{{{:.2f}}}"], 
"output_macros.tex")
`

where we have 2 sets of floats that we want to autofill in different formats. Let me know if that makes sense, thanks!

Reply:
Oh, got it. Thanks @jinfu93, that makes sense.

Would we want to test this case in [`test_output`](https://github.com/JMSLab/Template/blob/d85dbd8546607d213eea620763faa2792e518219/source/lib/JMSLab/tests/test_autofill.py#L33)?

Reply:
Great idea @santiagohermo! Added in [0eb794e](https://github.com/JMSLab/Template/pull/57/commits/0eb794ee6f892d4e2fdb6f3a6fad127f024f5624).

Reply:
Thanks @jinfu93! Just noting that we could both cases here.